src/resolver.rs

```diff
@@ -109,20 +112,30 @@ impl UpstreamUrlComponents {
         let now = Utc::now();
         let year_ago = now - Duration::try_days(DAYS_ONE_YEAR).unwrap();
 
+        // Parse cutoff_date if provided, otherwise use year_ago
+        let cutoff = self
+            .cutoff_date
+            .and_then(|date_str| {
+                chrono::NaiveDate::parse_from_str(&date_str, "%Y-%m-%d")
+                    .map(|date| date.and_hms_opt(0, 0, 0).unwrap().and_utc())
+                    .ok()
+            })
+            .unwrap_or(year_ago);
```

Comment:
```suggestion
            .unwrap_or_else(|| Utc::now() - Duration::try_days(DAYS_ONE_YEAR).unwrap());
```

I suppose we can lazily evaluate the default now.